### ISSUE TITLE
Non explicit error while loading a project

### ISSUE THREAD

**Issue by jecisc** [pharo-vcs/iceberg#625]
**OS:** Windows 10

**Pharo:** #60540

**Iceberg version:** v0.7.1

**What I did:**

I tried to load MDL in the image via this script:

```Smalltalk
Metacello new
    githubUser: 'DuneSt' project: 'MaterialDesignLite' commitish: 'v1.2.0' path: 'src';
    baseline: 'MaterialDesignLite';
    load
```

And I got this error:

```
UndefinedObject(Object)>>doesNotUnderstand: #locked
IceMetacelloDuplicatedNotification(MetacelloResolveProjectUpgrade)>>checkAllowed
IceMetacelloDuplicatedNotification(MetacelloAllowProjectUpgrade)>>defaultAction
UndefinedObject>>handleSignal:
Context>>handleSignal:
Context>>handleSignal:
IceMetacelloDuplicatedNotification(Exception)>>pass
[ ^ exception pass ] in MetacelloScriptEngine>>handleUpgrade: in Block: [ ^ exception pass ]
Dictionary>>at:ifAbsent:
MetacelloScriptEngine>>handleUpgrade:
```

**Comment by guillep** [comment]
Are you loading or upgrading MDL?

Was MDL loaded previously in your image? (even loaded and then manually removed?)

**Comment by jecisc** [comment]
I was loading MDL and it was not previously loaded.

**Comment by estebanlm** [comment]
I know why this happens. 
Is because there is already a version in disk and there is no onUpgrade: explicit in metacello.

**Comment by guillep** [comment]
So the solution is to use `onUpgrade:` ?

```
Metacello new
    baseline: #DiscordSt;
    repository: 'github://JurajKubelka/DiscordSt/src';
    onUpgrade: [ :e | e useIncoming ];
    load.
```

We should check why with metacello integration we get that  `self existingProjectRegistration == nil`

**Comment by guillep** [comment]
The main problem is fixed in commits b2e28c98f999dcfa5dff7c6416763a54916cb6d7 and dd3700ca470bd551a9d6092c363fb02ead60aaa8.

Then it is a matter of fixing #168
